cirq/ops/thermal_relaxation_channel.py

```diff
@@ -0,0 +1,267 @@
+# Copyright 2018 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Module for the thermal relaxation channel."""
+
+from typing import Iterable
+
+import numpy as np
+
+from cirq import protocols, value
+from cirq.ops import gate_features
+
+
+@value.value_equality
+class ThermalRelaxationChannel(gate_features.SingleQubitGate):
+    """Dampen qubits at varying rates through non ideal disipation.
+
+    This channel models the effect of energy dissipation into the environment
+    as well as the environment depositing energy into the system at varying
+    rates.
+    """
+
+    def __init__(self, p_exchange: float, p_relaxation: float,
+                 p_dephasing: float) -> None:
+        r"""The thermal relaxation channel.
+
+        Construct a channel to model energy dissipation into the environment
+        as well as the environment depositing energy into the system. The
+        probabilities with which the energy exchange occur are given by `gamma`
+        and `beta` with the probability of the environment being not excited is
+        given by `p`.
+
+        The stationary state of this channel is the diagonal density matrix
+        with probability `p` of being |0⟩ and probability `1-p` of being |1⟩.
```

Comment:
This is also true of the generalized amplitude damping channel. I'd at least point this out to avoid misleading the reader. Better yet, remove the comment and turn it into a unit test :-)

Reply:
Changed the docs.